src/libs/actions/IOU.js

```diff
@@ -530,7 +530,7 @@ function createDistanceRequest(report, participant, comment, created, transactio
             createdChatReportActionID,
             createdIOUReportActionID,
             reportPreviewReportActionID: reportPreviewAction.reportActionID,
-            waypoints: JSON.stringify(transaction.comment.waypoints),
+            waypoints: JSON.stringify(TransactionUtils.getValidWaypoints(transaction.comment.waypoints, true)),
```

Comment:
I don't think 2nd param is needed.
This will also be used in `GetRoute` api. And doesn't affect displaying routes.

Reply:
I think we need this logic to reorganize indices to be consecutive (i.e. we need to pass `true`)

Reply:
yes, I meant to reorganize always without needing to pass 2nd param but later I realized it should be conditional